src/plugins/vis_type_timeseries/server/lib/vis_data/helpers/get_bucket_size.js

```diff
@@ -65,14 +65,15 @@ const calculateBucketData = (timeInterval, capabilities) => {
   };
 };
 
-const calculateBucketSizeForAutoInterval = (req) => {
-  const duration = getTimerangeDuration(req);
+const calculateBucketSizeForAutoInterval = (req, maxBars) => {
+  const { from, to } = getTimerange(req);
+  const timerange = to.valueOf() - from.valueOf();
 
-  return calculateAuto.near(100, duration).asSeconds();
+  return calcAutoIntervalLessThan(maxBars, timerange).asSeconds();
```

Comment:
Okay so if I understand correctly this is the core part of the change, previously auto was hardcoded to 100, and now it's up to 100?

Reply:
Yes that true. Now we read `histogram:maxBars` configuration property as a limit value. The user can change it if necessary.